docs/architecture/architecture.md

```diff
@@ -0,0 +1,671 @@
+
+When handling the OCI `create` command, `cc-runtime` goes through the following steps:
+
+1. Create the networking container namespace on the host, according to the container
+  OCI configuration file. We only support networking namespaces for now, but
+  will support more of them later.
+2. Run all the [prestart OCI hooks](https://github.com/opencontainers/runtime-spec/blob/master/config.md#hooks)
+  in the host namespaces created in step 1, as described by the OCI container
+  configuration file.
+3. [Set up the container networking namespace up](#networking). This is when
+  all networking interfaces are created and setup inside the namespace, according
+  to the selected pod networking model (CNM for Docker or CNI for Kubernetes).
+4. Create and start the virtual machine running the container process. The VM
+  will run inside the host namespaces created during step 1, and its `systemd`
+  instance will spawn the `cc-agent` daemon.
+5. Register the virtual machine with `cc-proxy`.
+6. The `cc-proxy` waits for the agent to signal that it is ready and then returns
+  a token. This token uniquely identifies a process within a container inside
+  the virtual machine.
+7. Spawn the `cc-shim` process providing two arguments:
+  `cc-shim --token $(token) --uri $(uri)`
+   * The proxy URL, which can be either a UNIX or a TCP socket.
+   * The token for the container process it needs to monitor.
+8. The `cc-shim` connects to the proxy and signals which container process it
+  is going to monitor by passing its token through the `cc-proxy` `connectShim`
+  command.
+
+![Docker create](arch-images/create.png)
+
+At this point, the virtual machine that will run the containers workloads
+is up and running, and the [`cc-agent`](#agent) is ready to process container
+life cycle commands. The pod inside the virtual machine is not created, and
+the containers are not running yet. This will be done through the OCI 
+[`start`](#start) command
+
+#### [`start`](https://github.com/clearcontainers/runtime/blob/master/start.go)
+
+With namespace containers `start` launches a traditional Linux container process
+in its own set of namespaces. With Clear Containers, the main task of `cc-runtime`
+is to ask the [`cc-agent`](#agent) to create a pod inside the virtual machine
+and then start all containers within this pod.
+In practice, this means `cc-runtime` will run through the following steps:
+
+1. `cc-runtime` connects to `cc-proxy` and sends it the `attach` command to
+  let it know which VM we want to use to create a pod and start the new container.
+2. `cc-runtime` sends an agent `STARTPOD` command via `cc-proxy`. This creates
+  the container pod.
+3. `cc-runtime` sends an agent `NEWCONTAINER` command via `cc-proxy` in order to
+  create and start a new container in a given pod. The command is sent to `cc-proxy`
+  who forwards it to the right agent instance running in the appropriate guest.
+
```

Comment:
Just note - we don't mention the pause items here, but they appear in the state diagram.
It is also not clear if the container workload is actually executed by this command or not - we could add a small note to clarify that.